packages/cli/src/lib/Prompt.ts

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2020 Google Inc. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+import {Result} from '@bubblewrap/core';
+import * as inquirer from 'inquirer';
+
+export type ValidateFunction<T> = (input: string) => Result<T, Error>;
+
+export interface Prompt {
+  /**
+   * Prompts for free text input
+   * @param message a short description of the input
+   * @param defaultValue a default value or null
+   * @param convertFunction a function that converts from the input string to the return value.
+   * @param validateFunction an option function to validate the input.
+   */
+  promptInput<T>(
+    message: string,
+    defaultValue: string | null,
+    convertFunction: ValidateFunction<T>,
+  ): Promise<T>;
+
+  /**
+   * Prompts for free text input
+   * @param message a short description of the input
+   * @param choices a list of choices to be displayed to the user
+   * @param defaultValue a default value or null
+   * @param convertFunction a function that converts from the input string to the return value.
+   * @param validateFunction an option function to validate the input.
+   */
+  promptChoice<T>(
+    message: string,
+    choices: string[],
+    defaultValue: string | null,
+    convertFunction: ValidateFunction<T>,
+  ): Promise<T>;
+
+  /**
+   * Prompts for a password
```

Comment:
Could you add a comment about why this is different (eg, that the characters you type don't show up).

Reply:
Added more details to all methods.